src/main/java/org/broadinstitute/hellbender/tools/walkers/CombineGVCFs.java

```diff
@@ -0,0 +1,426 @@
+        headerLines.addAll(getDefaultToolVCFHeaderLines());
+
+        headerLines.addAll(annotationEngine.getVCFAnnotationDescriptions());
+
+        // add headers for annotations added by this tool
+        headerLines.add(VCFStandardHeaderLines.getInfoLine(VCFConstants.DEPTH_KEY));   // needed for gVCFs without DP tags
+        if ( dbsnp.dbsnp != null  ) {
+            VCFStandardHeaderLines.addStandardInfoLines(headerLines, true, VCFConstants.DBSNP_KEY);
+        }
+
+        VariantContextWriter writer = createVCFWriter(outputFile);
+
+        final Set<String> sampleNameSet = new IndexedSampleList(samples).asSetOfSamples();
+        final VCFHeader vcfHeader = new VCFHeader(headerLines, new TreeSet<>(sampleNameSet));
+        writer.writeHeader(vcfHeader);
+
+        return writer;
+    }
+
+    /**
+     * Method which calls endPreviousStates at the appropriate places on the given a new startingStates object
+     * and an OverallState object corresponding to the currently accumulated reads.
+     *
+     * @param variantContexts list of variant contexts with the same start position to be reduced
+     * @param referenceContext ReferenceContext object overlapping the provided VariantContexts
+     */
+    public void mergeWithnewVCs(final List<VariantContext> variantContexts, final ReferenceContext referenceContext) {
+        if ( !variantContexts.isEmpty() ) {
+            if ( ! okayToSkipThisSite(variantContexts, referenceContext) ) {
+                SimpleInterval loc = referenceContext.getInterval();
+                endPreviousStates( new SimpleInterval(loc.getContig(),loc.getStart()-1,loc.getStart()-1),
+                        Arrays.copyOfRange(referenceContext.getBases(), 1, referenceContext.getWindow().getLengthOnReference()),
+                        variantContexts,
+                        false);
+            }
+            VariantContextsOverlappingCurrentMerge.addAll(variantContexts);
+            for(final VariantContext vc : VariantContextsOverlappingCurrentMerge){
+                samples.addAll(vc.getSampleNames());
+            }
+        }
+    }
+
+    /**
+     * Is it okay to skip the given position?
+     *
+     * @param variantContexts  the query variant contexts representing the current position
+     * @param referenceContext  Reference context object overlapping the variant contexts
+     * @return true if it is okay to skip this position, false otherwise
+     */
+    private boolean okayToSkipThisSite(List<VariantContext> variantContexts, ReferenceContext referenceContext) {
+        Set<String> intersection = new HashSet<>(getSamples(variantContexts));
+        intersection.retainAll(samples);
+
+        //if there's a starting VC with a sample that's already in a current VC, don't skip this position
+        return prevPos != null && referenceContext.getInterval().getStart() == prevPos.getStart() + 1 && intersection.isEmpty(); //TODO off by 1 bug
```

Comment:
Leftover, but originally the concern was to do with changing the implementation from gatk3 where now i'm asking the reference context for the interval instead of from the reduced starting states